api/app/signals/apps/email_integrations/tests/test_rules.py

```diff
@@ -0,0 +1,220 @@
+# SPDX-License-Identifier: MPL-2.0
+# Copyright (C) 2021 - 2022 Gemeente Amsterdam
+from django.test import TestCase
+from factory.fuzzy import FuzzyText
+
+from signals.apps.email_integrations.rules import (
+    SignalCreatedRule,
+    SignalHandledRule,
+    SignalOptionalRule,
+    SignalReactionRequestReceivedRule,
+    SignalReactionRequestRule,
+    SignalReopenedRule,
+    SignalScheduledRule
+)
+from signals.apps.questionnaires.app_settings import NO_REACTION_RECEIVED_TEXT
+from signals.apps.signals import workflow
+from signals.apps.signals.factories import SignalFactory, StatusFactory
+
+
+class RuleTestMixin:
+    rule = None
+    state = None
+
+    def test_happy_flow(self):
+        status_text = FuzzyText(length=200) if self.state == workflow.REACTIE_GEVRAAGD else FuzzyText(length=400)
+        signal = SignalFactory.create(status__state=self.state, status__text=status_text,
+                                      reporter__email='test@example.com')
+        self.assertTrue(self.rule(signal))
+
+    def test_signal_not_happy_flow(self):
```

Comment:
What is the intention here? Does this assume `workflow.BEHANDELING` is not allowed to send emails ever?

Reply:
Old test that should have been removed.